packages/fuels/tests/logs.rs

```diff
@@ -439,5 +441,187 @@ async fn test_multi_call_require_log_multi_contract() -> Result<(), Error> {
+    let expected_deeply_nested_struct = StructDeeplyNestedGeneric {
+        field_1: expected_nested_struct.clone(),
+        field_2: 64,
+    };
+
+    let log_u64 = response.get_logs_with_type::<u64>()?;
+    let log_u32 = response.get_logs_with_type::<u32>()?;
+    let log_u16 = response.get_logs_with_type::<u16>()?;
+    let log_u8 = response.get_logs_with_type::<u8>()?;
+    let log_struct = response.get_logs_with_type::<TestStruct>()?;
+    let log_enum = response.get_logs_with_type::<TestEnum>()?;
+    let log_generic_struct = response.get_logs_with_type::<StructWithGeneric<TestStruct>>()?;
+    let log_generic_enum = response.get_logs_with_type::<EnumWithGeneric<[_; 3]>>()?;
+    let log_nested_struct =
+        response.get_logs_with_type::<StructWithNestedGeneric<StructWithGeneric<TestStruct>>>()?;
+    let log_deeply_nested_struct = response.get_logs_with_type::<StructDeeplyNestedGeneric<
+        StructWithNestedGeneric<StructWithGeneric<TestStruct>>,
+    >>()?;
+    // try to retrieve non existent log
+    let log_nonexistent = response.get_logs_with_type::<bool>()?;
+
+    assert_eq!(log_u64, vec![128, 64]);
+    assert_eq!(log_u32, vec![32]);
+    assert_eq!(log_u16, vec![16]);
+    assert_eq!(log_u8, vec![8]);
+    assert_eq!(log_struct, vec![expected_struct]);
+    assert_eq!(log_enum, vec![expected_enum]);
+    assert_eq!(log_generic_struct, vec![expected_generic_struct]);
+    assert_eq!(log_generic_enum, vec![expected_generic_enum]);
+    assert_eq!(log_nested_struct, vec![expected_nested_struct]);
+    assert_eq!(
+        log_deeply_nested_struct,
+        vec![expected_deeply_nested_struct]
+    );
+    assert!(log_nonexistent.is_empty());
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_script_require_log() -> Result<(), Error> {
+    script_abigen!(
+        log_script,
+        "packages/fuels/tests/scripts/script_require/out/debug/script_require-abi.json"
+    );
+
+    let wallet = launch_provider_and_get_wallet().await;
+    let bin_path = "../fuels/tests/scripts/script_require/out/debug/script_require.bin";
+    let instance = log_script::new(wallet.clone(), bin_path);
+
+    {
+        let error = instance
+            .main(MatchEnum::RequirePrimitive())
+            .call()
+            .await
+            .expect_err("Should return a revert error");
+
+        assert_is_revert_containing_msg("42", error);
+    }
+    {
+        let error = instance
+            .main(MatchEnum::RequireString())
+            .call()
+            .await
+            .expect_err("Should return a revert error");
+
+        assert_is_revert_containing_msg("fuel", error);
+    }
+    {
+        let instance = log_script::new(wallet.clone(), bin_path);
+        let error = instance
+            .main(MatchEnum::RequireCustomGeneric())
+            .call()
+            .await
+            .expect_err("Should return a revert error");
+
+        assert_is_revert_containing_msg("StructDeeplyNestedGeneric", error);
+    }
+
     Ok(())
```

Comment:
Those tests are really good

Reply:
I have to give credit to @MujkicA. Those are his test but modified for scripts :smile: